src/Linux/local_cache.cpp

```diff
@@ -21,7 +21,11 @@ constexpr uint16_t CACHE_V1 = 1;
 
 constexpr locale_t NULL_LOCALE = reinterpret_cast<locale_t>(0);
 
-static std::string g_cache_dirname;
+static std::string g_cache_dirname = "";
+static std::mutex cache_directory_lock;
+
+static constexpr size_t CACHE_LOCATIONS = 5;
+static const char *cache_locations[CACHE_LOCATIONS];

```

Comment:
I think the code dealing with cache_locations would be a little safer and shorter if it was a vector of strings.